### PR TITLE
feat: Implement Jenkins models

### LINKED ISSUE #1076: Config Validation: Jenkins - implement models required for validation
**Description**
Jenkins - implement models required for validation

**Acceptance Criteria**
- [ ]  All models required for Jenkins Source Config Validation are implemented.
- [ ] Test all models.

**Additional context**
[CI Integrations Config Validator](https://github.com/platform-platform/monorepo/blob/master/metrics/ci_integrations/docs/04_ci_integrations_config_validator.md)

### PR DESCRIPTION
**Pull request by solid-yevheniiashchypeleva**
Fixes #1076

## Checklist

- [x] Documented
- [x] Unit tested

### CODE REVIEW

**Comment by solid-yevheniiashchypeleva** [comment]
I have read the CLA Document and I hereby sign the CLA

File: metrics/ci_integrations/lib/client/jenkins/model/jenkins_instance_info.dart [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    return JenkinsInstanceInfo(
+      version: json['X-Jenkins'] as String,
+    );
+  }
+
+  /// Creates a list of [JenkinsInstanceInfo]s from the given [list] of decoded
+  /// JSON objects.
+  ///
+  /// Returns `null` if the given [list] is `null`.
+  static List<JenkinsInstanceInfo> listFromJson(List<dynamic> list) {
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/lib/client/jenkins/model/jenkins_instance_info.dart:34]: Do we actually need this method here?

File: metrics/ci_integrations/test/client/jenkins/model/jenkins_instance_info_test.dart [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+    test(
+      ".toJson() converts an instance to the json encodable map",
+      () {
+        final json = instanceInfo.toJson();
+
+        expect(json, equals(instanceInfoJson));
+      },
+    );
+  });
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/test/client/jenkins/model/jenkins_instance_info_test.dart:87]: Should we test the `toString` method override? The same suggestions for the `JenkinsUser` class.

**Review by solid-vovabeloded — CHANGES_REQUESTED** [review]
Nicely done, @solid-daniyilyevtyushkin & @solid-yevheniiashchypeleva! I've added a few suggestions here - please take a look. Overall, the code looks great - thanks!

**Review by solid-vovabeloded — APPROVED** [review]
LGTM, thanks @solid-daniyilyevtyushkin & @solid-yevheniiashchypeleva!

**Pull request merged**